src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -108,6 +108,20 @@ public static void BeginInvokeDotNet(JSRuntime jsRuntime, DotNetInvocationInfo i
             // Returned a task - we need to continue that task and then report an exception
             // or return the value.
             task.ContinueWith(t => EndInvokeDotNetAfterTask(t, jsRuntime, invocationInfo), TaskScheduler.Current);
+
+        }
+        else if (syncResult?.GetType().GetGenericTypeDefinition() == typeof(ValueTask<>))
+        {
+            //since syncResult is generic we have to get AsTask method for its type
+            var valueTaskAsTaskMethod = syncResult.GetType().GetMethod("AsTask", BindingFlags.Instance | BindingFlags.Public);
```

Comment:
On the second thought, is there any way to achieve this without calling the `AsTask`? I think calling it would defeat the whole purpose of using `ValueTask` on the return type.